code/core/src/common/js-package-manager/JsPackageManager.ts

```diff
@@ -134,88 +127,43 @@ export abstract class JsPackageManager {
     }
   }
 
-  packageJsonPath(): string {
-    if (!this.cwd) {
-      throw new Error('Missing cwd');
-    }
-    return resolve(this.cwd, 'package.json');
-  }
-
-  async readPackageJson(): Promise<PackageJson> {
-    const packageJsonPath = this.packageJsonPath();
-    if (!existsSync(packageJsonPath)) {
-      throw new Error(`Could not read package.json file at ${packageJsonPath}`);
-    }
+  /** Read the `package.json` file available in the provided directory */
+  static getPackageJson(packageJsonPath: string): PackageJsonWithDepsAndDevDeps {
```

Comment:
same question regarding getPackageJson/writePackageJson, do they need to be sync?